utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -0,0 +1,281 @@
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.initMocks(this)
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setTotalMemory_returnsCorrectMemoryTier() {
+    val activityManager: ActivityManager =
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    val shadowActivityManager: ShadowActivityManager = shadowOf(activityManager)
+    val memoryInfo = ActivityManager.MemoryInfo()
+    memoryInfo.totalMem = (1.5 * 1024 * 1024 * 1024).toLong()
+    shadowActivityManager.setMemoryInfo(memoryInfo)
+    val memoryTier = performanceMetricsUtilsImpl.getDeviceMemoryTier()
+
+    assertThat(memoryTier).isEqualTo(OppiaMetricLog.MemoryTier.LOW_MEMORY_TIER)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getTotalStorageUsed_returnsCorrectStorageUsage() {
+    context.openFileOutput(TEST_FILE_NAME, Context.MODE_PRIVATE).use {
+      it.write(TEST_FILE_CONTENT.toByteArray())
+    }
+    val cacheFile = File.createTempFile(TEST_APP_PATH, null, context.cacheDir)
+    val expectedStorageValue = TEST_FILE_CONTENT.length + cacheFile.length()
+
+    assertThat(performanceMetricsUtilsImpl.getUsedStorage()).isEqualTo(expectedStorageValue)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getTotalStorageUsageTier_returnsCorrectStorageUsageTier() {
+    context.openFileOutput(TEST_FILE_NAME, Context.MODE_PRIVATE).use {
+      it.write(TEST_FILE_CONTENT.toByteArray())
+    }
+    val cacheFile = File.createTempFile(TEST_APP_PATH, null, context.cacheDir)
+    val expectedStorageValue = TEST_FILE_CONTENT.length + cacheFile.length()
+
+    val expectedStorageTierValue = when (expectedStorageValue / (1024 * 1024 * 1024)) {
+      in 0L until 32L -> OppiaMetricLog.StorageTier.LOW_STORAGE
+      in 32L..64L -> OppiaMetricLog.StorageTier.MEDIUM_STORAGE
+      else -> OppiaMetricLog.StorageTier.HIGH_STORAGE
+    }
+
+    assertThat(performanceMetricsUtilsImpl.getDeviceStorageTier())
+      .isEqualTo(expectedStorageTierValue)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getBytesSent_returnsCorrectAmountOfNetworkBytesSent() {
+    val expectedNetworkBytesSent = TrafficStats.getUidTxBytes(
+      ApplicationProvider.getApplicationContext<Application>().applicationInfo.uid
+    )
+
+    assertThat(performanceMetricsUtilsImpl.getTotalSentBytes())
+      .isEqualTo(expectedNetworkBytesSent)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getBytesReceived_returnsCorrectAmountOfNetworkBytesReceived() {
+    val expectedNetworkBytesReceived = TrafficStats.getUidRxBytes(
+      ApplicationProvider.getApplicationContext<Application>().applicationInfo.uid
+    )
+
+    assertThat(performanceMetricsUtilsImpl.getTotalReceivedBytes())
+      .isEqualTo(expectedNetworkBytesReceived)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setAppProcesses_getMemoryUsage_returnsCorrectMemoryUsage() {
```

Comment:
For this and the two above, we ought to be verifying specific values to ensure that we're actually testing the logic of the implementation (to guard against the case, for example, where these methods just return 0 which could allow a broken implementation to pass the tests),

Reply:
done.